Add migration note about Provider discovery via the Service Loader (#25324)
quarkusio/quarkus: docs/src/main/asciidoc/resteasy-reactive-migration.adoc

The most common case where the difference in providers might result in different behavior, is the included `javax.ws.rs.ext.ExceptionMapper` implementations. To see what classes are included in the application, launch the application in dev mode and navigate to http://localhost:8080/q/dev/io.quarkus.quarkus-resteasy-reactive/exception-mappers.

==== Service Loading

RESTEasy Classic supports determining providers at build time using Java's Service Loader. In order to ensure that all providers are determined at build time,

> RESTEasy Reactive does not support this feature. Instead, users that have providers in application dependencies all encouraged to make index those dependencies

Reviewer: ```suggestion
RESTEasy Reactive does not support this feature. Instead, users that have providers in application dependencies are encouraged to index those dependencies
```
